types/src/front/api_handlers/internal/agent_configuration.ts

```diff
@@ -202,3 +202,7 @@ export const PostOrPatchAgentConfigurationRequestBodySchema = t.type({
 export type PostOrPatchAgentConfigurationRequestBody = t.TypeOf<
   typeof PostOrPatchAgentConfigurationRequestBodySchema
 >;
+
+export type AgentActionConfigurationType = t.TypeOf<
```

Comment:
You're clashing with the one in agent.ts of types no?

Reply:
I'm still bearish on using that type in lib/api/... since it's an iots dependency and we should not depend on iots in lib/api/xxx IMHO

Reply:
So then where would you put this function? I agree on functions in `/lib/api` to not return `APIErrorResponse` since those are meant to be used on other parts as well. However, body payload to create/update a resource are acceptable IMHO.

Reply:
I think we should really stay away from them and have lib/api/... operate on explicit types we define. iots is meant for parsing in no way it is an accepeted way to define types.

What's the difrerence between AgentActionConfigurationType and RawAgentActionConfigurationType (the very fact that we have that Raw prefix shows that it's not really principled).

Reply:
I mean `AgentActionConfigurationType` enforce the presence of `id` and `sId` so we need a type that represent those actions that have not been saved yet. I'm fine to derive this type from the `AgentActionConfigurationType` if that makes your more comfortable. I'll stick to my point that we need to be able to represent this concept in `lib/api` still.

Reply:
Updated to use the app types instead of io-ts. There are some pre-multi actions types, created [this](https://github.com/dust-tt/dust/issues/8419) to rework.